src/main/ank/2.3-free-monoids.md

```diff
@@ -0,0 +1,87 @@
+```Haskell
+class Monoid m where
+  mempty :: m
+  mappend :: m -> m -> m
+```
+```kotlin:ank:silent
+interface Monoid<M> {
+  fun mempty() : M
+  fun mappend(a1: M, a2: M): M
+}
+```
+................
+```Haskell
+instance Monoid [a] where
+  mempty = []
+  mappend = (++)
+```
+```kotlin:ank:silent
+class ListMonoid<M> : Monoid<List<M>> {
```

Comment:
Use `A` as a type arg instead of `M` since that is the Arrow convention for concrete not kinded type args